MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -1914,6 +1970,22 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
                 filters.Add("hwupload");
             }
 
+            var videoDecoder = GetHardwareAcceleratedVideoDecoder(state, options);
+
+            // If we are software decoding, and hardware encoding	  
+            if (string.Equals(outputVideoCodec, "h264_qsv", StringComparison.OrdinalIgnoreCase) 
+                && (string.IsNullOrEmpty(videoDecoder) || !videoDecoder.Contains("qsv")))
+            {
+                filters.Add("format=nv12|qsv");
+                filters.Add("hwupload=extra_hw_frames=64");
+            }
+
+            var inputWidth = videoStream?.Width;
+            var inputHeight = videoStream?.Height;
+            var threeDFormat = state.MediaSource.Video3DFormat;
+
+            filters.AddRange(GetScalingFilters(inputWidth, inputHeight, threeDFormat, videoDecoder, outputVideoCodec, request.Width, request.Height, request.MaxWidth, request.MaxHeight));
```

Comment:
why did you move these?

Reply:
fixed. it was moved while working out what  the code does and i forgot to move it back afterwards.